pallets/cash/src/lib.rs

```diff
@@ -207,6 +207,12 @@ decl_storage! {
         /// Miner of the current block.
         Miner get(fn miner): Option<ChainAccount>;
 
+        /// Mapping of total principal paid to each miner.
+        MinerCumulative get(fn miner_cumulative): map hasher(blake2_128_concat) ChainAccount => CashPrincipalAmount;
+
+        /// Total principal paid to all miners.
+        TotalMinerCumulative get(fn total_miner_cumulative): CashPrincipalAmount;
```

Comment:
would it be better to iterate MinerCumulative in rpc's and then not need this storage?

Reply:
That's probably excessively inefficient? I mean no worse the liquidity endpoint, but we should try to minimize/eliminate such endpoints I think...I think real answer longer term is events and apis (subscan, subquery should actually be able to handle this with proper eventing)

Reply:
it would be like... 13 reads in a dedicated rpc node, a single trx request probably has more than that many reads in it.

Reply:
also we'll be wanting each miner's cumulative to be up to date, so those per-validator reads will be happening anyways

Reply:
or we could just write TotalMinerCumulative anyways 🤷